Update test matrix

SergiiDmytruk opened the pull request:
* Add Dell OptiPlex 9010, which was tested, but not documented
* Add a new entry for Asus KGPE-D16 because of a different TPM version and firmware (don't know, maybe TPM version was a typo).
* Document some hardware-specific information in a separate table.

SergiiDmytruk committed:
test_matrix: add Dell OptiPlex 9010 and Asus KGPE-D16

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
test_matrix: document known hardware-specific issues

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

pietrushnic commented on docs/documentation/test_matrix.md:
```diff
@@ -7,8 +7,22 @@ Devices and configurations TrenchBoot has been tested on.
 | Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  PASS  | CSM legacy boot |
-| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  FAIL  | UEFI boot |
\ No newline at end of file
+| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  FAIL  | UEFI boot |
```
Do we have bug tracking for this failure?

SergiiDmytruk replied:
I don't see it in TrenchBoot organization, so maybe not.  This is several years old, don't know what it's about.  Maybe @miczyg1 or @krystian-hebel have some pointers where additional information can be found.

miczyg1 replied:
UEFI boot of Xen was not supported and still isn't. That is the most likely reason of FAIL here.

Not sure if I have ever tested the Supermicro platform by booting TrenchBooted Linux.

pietrushnic replied:
I would appreciate it if you could create an issue since we reported failure. If no one can refer to what failure we have here then maybe we should remove those lines. But that would mean losing potentially useful information. @macpijan @stacktrust @dpsmith any opinions here?

SergiiDmytruk replied:
See b1c744ea2b347ff114b70cb5656b42b7b62fb863

pietrushnic replied:
@SergiiDmytruk thank you

pietrushnic commented on docs/documentation/test_matrix.md:
```diff
@@ -7,8 +7,22 @@ Devices and configurations TrenchBoot has been tested on.
+| (continued)                                                                  | 2. Xen sometimes has issues rebooting, boot cycle is the workaround. |
+| Asus KGPE-D16 <br>(AMD Opteron family 15h models 00h-0fh server)             | IOMMU has no extended features: can't use `INVALIDATE_IOMMU_ALL` in SKL. |
+| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   | Problematic USB controller for Qubes OS ([resets the system][qubesos-m11-reset]). |
+| (continued)                                                                  | Works without `sys-usb` VM or if USB controller is disabled. |
```
I think all of those should have tracking issues associated. WDYT @stacktrust @dpsmith ?

stacktrust replied:
Yes, each test failure can be linked to a tracking issue.

Current table should be separated into two tables, one for tested-good (PASS) hardware.  Add a column to that table with sales status, e.g. YYYY-YYYY (year first sold, year for End of Sale).  Currently, everything in the table is probably no longer sold.

The table with test failures can be divided into two:  hardware that can never be supported due to an unchangeable limitation, or hardware that could potentially be supported by a firmware or software (OS, driver) fix.

Somewhere on this page should be a description or link to the tests which map onto "PASS" status, e.g. is this related to fwupd hardware test version ABC? AEM-with-TrenchBoot version XYZ?

SergiiDmytruk replied:
These aren't test failures of TrenchBoot/AEM, these are difficulties/things of note one has to face when using these platforms today. They were probably more usable years ago, but something has changed in software that nobody tested on these devices.

OptiPlex, M11 and KGPE-D16 work with TrenchBoot/AEM.  Don't know about older entries and how they were tested.

Split first table in 50f70cb and added availability column (seems to be possible to buy new Supermicro M11).

pietrushnic replied:
@SergiiDmytruk I'm on mobile so can't check it but what do blame to say about the author who initially added those entries with failures? I don't want to hold this merge but OTOH we should stop adding vague failure statements without any meaningful references. Who knows when next time this file will be updated?

SergiiDmytruk replied:
It was added by @miczyg1 in 3a3a16d, which is why I mentioned him in another comment, the files in that import were supposedly moved from https://github.com/TrenchBoot/documentation/ repo, but I didn't find the test matrix there.

I now think that these are actual CI results related to https://github.com/3mdeb/meta-trenchboot like https://gitlab.com/trenchboot1/3mdeb/meta-trenchboot/-/pipelines/173780864.  https://github.com/3mdeb/trenchboot-ci could also be related.  This looks like Debian/yocto build with Trenchboot, not AEM/Xen/QubesOS like new test results.

pietrushnic replied:
So, maybe we should explain what we mean by "tested" or "known to work" for the entries that we add now. For entries for which we don't know how testing was done or what the failure reason was, let's create an issue to investigate and find out and link that issue here so anyone who stumbles on it will poster us for an answer. Hopefully @BeataZdunczyk can allocate some resources to fix that in further phases.

I really miss transparent validation here.

SergiiDmytruk replied:
See 382a853954aa6f18db8b28e9a3b42e124b058f2f and b1c744ea2b347ff114b70cb5656b42b7b62fb863

pietrushnic replied:
@SergiiDmytruk thank you, it helps a lot but is still not up to the transparent validation standard since OpenQA is vaguely mentioned.  Can we point to specific test reports maybe GitHub issues where the test results were published? So when someone would like to ask what was really tested and how, it would be answered. Also if we tested that manually and someone personally confirmed it work maybe methodology was described in some place and we can say methodology described here (link) was tested by Joe Doe at version X and he confirmed it works.

I hope you understand my point. I just want to avoid similar issues we face right now with hard to identify test results. Transparent validation is the key trustworthiness ingredient.

SergiiDmytruk replied:
See dae0be6bc5b1ea2e9fbebcc8eb4c8deb43f9d064.  Test results for openQA are in the history of a non-public openQA instance.  Outcome of manual test results aren't documented in a single place, mostly comments in PRs, issues or Jira.  I've linked to openQA tests distribution that's being used, explained steps that both automatic and manual testing involves and linked to blog posts which explain the process in more detail (with commands and screenshots).

pietrushnic replied:
Thank you. That resolve both conversations IMHO.

SergiiDmytruk committed:
test_matrix: separate working from non-working devices

Also add availability column to the table of working ones.  Release year
is sometimes clearly stated or can be derived from news/reviews, while
end of production might not be declared anywhere and thus can be
approximate.

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

krystian-hebel commented on docs/documentation/test_matrix.md:
```diff
@@ -1,14 +1,34 @@
-| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  PASS  | CSM legacy boot |
-| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  FAIL  | UEFI boot |
\ No newline at end of file
+Devices and configurations on which TrenchBoot is known to work (availability
```
I would also add [HP Thin Client t630](https://support.hp.com/us-en/document/c05240287) to this list. Except problem with starting original ISO in legacy mode (similar to https://github.com/QubesOS/qubes-issues/issues/8462#issuecomment-1697323478, but it only worked for me after i also changed partition type from protective MBR to Linux, in addition to marking it active), everything works out of the box. I'm not sure if it can be bought as new, but [BIOS was updated two months ago](https://support.hp.com/us-en/drivers/hp-t630-thin-client/10522151) so at least it is still supported by vendor.

![image](https://github.com/TrenchBoot/TrenchBoot.github.io/assets/40995177/520b10f8-5007-4fd2-be4d-10d39dd9e208)

SergiiDmytruk replied:
Added in ea216986222c99ad18fbbcd0f74342fded899008 linking to your comment.

SergiiDmytruk committed:
test_matrix: add HP Thin Client t630

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

pietrushnic reviewed (changes requested):
@BeataZdunczyk if we can't resolve issues in expected timeline I'm OK with resolving those based on separate issues which you will commit to address in later phases of the project.

SergiiDmytruk committed:
test_matrix: improve hardware quirks and workarounds section

Explain the purpose of the section.

Sort list of devices alphabetically.

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
test_matrix: split results into old and new

Not all hardware was tested in the same way, so don't put it all
together.

Also sort devices alphabetically.

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
test_matrix: link to issues for failures

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
test_matrix: explain modern testing results in more detail

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
